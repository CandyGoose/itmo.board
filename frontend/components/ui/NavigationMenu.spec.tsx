import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import {
    NavigationMenu,
    NavigationMenuList,
    NavigationMenuItem,
    NavigationMenuTrigger,
    NavigationMenuContent,
    NavigationMenuLink,
    NavigationMenuIndicator,
    NavigationMenuViewport,
} from './NavigationMenu';

const MockMenu = () => (
    <NavigationMenu>
        <NavigationMenuList>
            <NavigationMenuItem>
                <NavigationMenuTrigger>Menu 1</NavigationMenuTrigger>
                <NavigationMenuContent>
                    <NavigationMenuLink href="/option1">
                        Option 1
                    </NavigationMenuLink>
                    <NavigationMenuLink href="/option2">
                        Option 2
                    </NavigationMenuLink>
                </NavigationMenuContent>
            </NavigationMenuItem>
            <NavigationMenuItem>
                <NavigationMenuTrigger>Menu 2</NavigationMenuTrigger>
                <NavigationMenuContent>
                    <NavigationMenuLink href="/option3">
                        Option 3
                    </NavigationMenuLink>
                    <NavigationMenuLink href="/option4">
                        Option 4
                    </NavigationMenuLink>
                </NavigationMenuContent>
            </NavigationMenuItem>
        </NavigationMenuList>
        <NavigationMenuIndicator />
        <NavigationMenuViewport />
    </NavigationMenu>
);

describe('NavigationMenu Component', () => {
    test('renders NavigationMenu correctly', () => {
        render(<MockMenu />);

        expect(screen.getByText('Menu 1')).toBeInTheDocument();
        expect(screen.getByText('Menu 2')).toBeInTheDocument();
    });

    test('renders without crashing', () => {
        render(<MockMenu />);
        expect(screen.getByText('Menu 1')).toBeInTheDocument();
    });
});
