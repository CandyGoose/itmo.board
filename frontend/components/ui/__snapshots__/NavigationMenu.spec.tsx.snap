// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu Components NavigationMenu renders without crashing 1`] = `
<div>
  <nav
    aria-label="Main"
    class="relative z-10 flex max-w-max flex-1 items-center justify-center"
    data-orientation="horizontal"
    dir="ltr"
  >
    <div>
      Menu Content
    </div>
    <div
      class="absolute left-0 top-full flex justify-center"
    />
  </nav>
</div>
`;
